### PR TITLE
[MI-2629] Add feature to edit filters of the subscription

### PR DESCRIPTION
**Pull request by ayusht2810**
[demo](https://user-images.githubusercontent.com/100013900/213220915-346d3505-f069-4142-8ae9-f0baeba185e5.webm)

### CODE REVIEW

File: webapp/src/containers/addOrEditSubscriptions/subComponents/filtersPanel.tsx [hunk 1]
```diff
@@ -149,6 +167,50 @@ const FiltersPanel = forwardRef<HTMLDivElement, FiltersPanelProps>(({
...
+                };
+
+                if (filter.filterType === SupportedFilters.ASSIGNMENT_GROUP) {
+                    setGetAssignmentGroupPayload(payload);
+                } else {
+                    setGetServicePayload(payload);
+                }
+
+                makeApiRequestWithCompletionStatus(Constants.pluginApiServiceConfigs.getFilterData.apiServiceName, payload);
+                return '';
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/addOrEditSubscriptions/subComponents/filtersPanel.tsx:185]: need for this empty return ?

**Review by Kshitij-Katiyar — APPROVED** [review]

File: webapp/.eslintrc.json [hunk 2]
```diff
@@ -188,7 +188,7 @@
     "max-lines": [
       1,
       {
-        "max": 500,
+        "max": 600,
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/.eslintrc.json:191]: the lines of the file were more than this, which was giving eslint warnings, so updated them

File: webapp/src/containers/addOrEditSubscriptions/subComponents/filtersPanel.tsx (hunk 1 above)
Comment (ayusht2810) [reply webapp/src/containers/addOrEditSubscriptions/subComponents/filtersPanel.tsx:185]: was getting eslint error `Array.prototype.map() expects a return value from arrow function`

File: webapp/.eslintrc.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kshitij-Katiyar) [reply webapp/.eslintrc.json:191]: can we segregate those lines? I mean we usually should not change the actual rules.

**Review by Kshitij-Katiyar — APPROVED** [review]

File: webapp/.eslintrc.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/.eslintrc.json:191]: I don't think changing this rule will have any problem. What do you think @manojmalik20 @avas27JTG ?

File: webapp/src/containers/addOrEditSubscriptions/subComponents/filtersPanel.tsx [hunk 3]
```diff
@@ -127,6 +140,11 @@ const FiltersPanel = forwardRef<HTMLDivElement, FiltersPanelProps>(({
         return {isLoading, data: data as FieldsFilterData[], isError, error: (apiErr as FetchBaseQueryError)?.data as APIError | undefined};
     };
 
+    const getItems = (payload: SearchFilterItemsParams | null) => {
+        const {isLoading, data: getItemsData, isError, isSuccess, error: apiErr} = getApiState(Constants.pluginApiServiceConfigs.getFilterData.apiServiceName, payload);
+        return {isLoading, data: getItemsData as FieldsFilterData[], isError, isSuccess, error: (apiErr as FetchBaseQueryError)?.data as APIError | undefined};
```
Comment (manojmalik20) [inline webapp/src/containers/addOrEditSubscriptions/subComponents/filtersPanel.tsx:145]:
```suggestion
        return {isLoading, data: getItemsData as FilterFieldsData[], isError, isSuccess, error: (apiErr as FetchBaseQueryError)?.data as APIError | undefined};
```
I think `FilterFieldsData` is more appropriate name for the type.

File: webapp/src/utils/index.tsx [hunk 4]
```diff
@@ -175,6 +175,19 @@ const getFormattedFilters = (filters: FiltersData[]) => {
...
+    if (!formattedFilters) {
+        return [];
+    }
+
+    const filters: FiltersData[] = [];
+    JSON.parse(formattedFilters, (filterType, filterValue) => {
+        filters.push({filterType, filterValue});
+    });
+
+    return filters.slice(0, -1);
```
Comment (manojmalik20) [inline webapp/src/utils/index.tsx:188]: Why are you removing the last element?

File: webapp/.eslintrc.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (manojmalik20) [reply webapp/.eslintrc.json:191]: I don't think this should be an issue.

File: webapp/src/utils/index.tsx (hunk 4 above)
Comment (ayusht2810) [reply webapp/src/utils/index.tsx:188]: to remove an extra element that is added to the list

**Review by manojmalik20 — APPROVED** [review]

File: webapp/src/utils/index.tsx (hunk 4 above)
Comment (manojmalik20) [reply webapp/src/utils/index.tsx:188]: What is the extra element that is added to the list?

**Review by manojmalik20 — APPROVED** [review]

File: webapp/src/utils/index.tsx (hunk 4 above)
Comment (ayusht2810) [reply webapp/src/utils/index.tsx:188]: it is removed in the following PR and the element is key-value pair of `""`, `{}`

**Pull request merged**